src/frontend/A32/translate/impl/asimd_three_same.cpp

```diff
@@ -57,6 +65,79 @@ bool FloatingPointInstruction(ArmTranslatorVisitor& v, bool D, bool sz, size_t V
     v.ir.SetVector(d, result);
     return true;
 }
+
+bool IntegerComparison(ArmTranslatorVisitor& v, bool U, bool D, size_t sz, size_t Vn, size_t Vd, bool N, bool Q, bool M, size_t Vm,
+                       Comparison comparison) {
+    if (sz == 0b11) {
+        return v.UndefinedInstruction();
+    }
+
+    if (Q && (Common::Bit<0>(Vd) || Common::Bit<0>(Vn) || Common::Bit<0>(Vm))) {
+        return v.UndefinedInstruction();
+    }
+
+    const size_t esize = 8 << sz;
+    const auto d = ToVector(Q, Vd, D);
+    const auto m = ToVector(Q, Vm, M);
+    const auto n = ToVector(Q, Vn, N);
+
+    const auto reg_n = v.ir.GetVector(n);
+    const auto reg_m = v.ir.GetVector(m);
+    const auto result = [&] {
+        switch (comparison) {
+        case Comparison::GT:
+            return U ? v.ir.VectorGreaterUnsigned(esize, reg_n, reg_m)
+                     : v.ir.VectorGreaterSigned(esize, reg_n, reg_m);
+        case Comparison::GE:
+            return U ? v.ir.VectorGreaterEqualUnsigned(esize, reg_n, reg_m)
+                     : v.ir.VectorGreaterEqualSigned(esize, reg_n, reg_m);
+        case Comparison::EQ:
+            return v.ir.VectorEqual(esize, reg_n, reg_m);
+        default:
+            return IR::U128{};
+        }
+    }();
+
+    v.ir.SetVector(d, result);
+    return true;
+}
+
+bool FloatComparison(ArmTranslatorVisitor& v, bool D, bool sz, size_t Vn, size_t Vd, bool N, bool Q, bool M, size_t Vm,
+                     Comparison comparison) {
+    if (sz) {
+        return v.UndefinedInstruction();
+    }
+
+    if (Q && (Common::Bit<0>(Vd) || Common::Bit<0>(Vn) || Common::Bit<0>(Vm))) {
+        return v.UndefinedInstruction();
+    }
+
+    const auto d = ToVector(Q, Vd, D);
+    const auto m = ToVector(Q, Vm, M);
+    const auto n = ToVector(Q, Vn, N);
+
+    const auto reg_n = v.ir.GetVector(n);
+    const auto reg_m = v.ir.GetVector(m);
+    const auto result = [&] {
+        switch (comparison) {
+        case Comparison::GE:
+            return v.ir.FPVectorGreaterEqual(32, reg_n, reg_m);
```

Comment:
```suggestion
            return v.ir.FPVectorGreaterEqual(32, reg_n, reg_m, false);
```

And the same for alll the other FP calls except for FPVectorAbs.

Reply:
Done